### DISCUSSION TITLE
Why in the `.Only()` methods there is `.Limit(2)`?

### DISCUSSION THREAD

**Discussion by frederikhors** [ent/ent#2602] [Q&A] [answered]
Why in the `.Only()` methods there is `.Limit(2)`?

I would expect `Limit(1)`.

**Comment by rotemtam** [comment, marked as answer]
Hey

Look at the implementation for `Only`:
https://github.com/ent/ent/blob/e1c5277483cc0f725ea44044ce11624af19cb0a1/entc/gen/template/builder/query.tmpl#L148-L164

It retrieves two records to determine if there are more than one in which case an error isr returned.

**Comment by a8m** [comment]
If you don’t care about this check, use First/FirstX instead.
